yarn-project/cli-wallet/src/cmds/deploy.ts

```diff
@@ -40,9 +40,17 @@ export async function deploy(
       `\nWarning: Contract was compiled with a different version of Aztec.nr: ${contractArtifact.aztecNrVersion}. Consider updating Aztec.nr to ${expectedAztecNrVersion}\n`,
     );
   }
-  const { getSchnorrAccount } = await import('@aztec/accounts/schnorr');
 
-  const wallet = await getSchnorrAccount(client, privateKey, deriveSigningKey(privateKey), Fr.ZERO).getWallet();
+  let wallet;
+  if (aliasOrAddress) {
+    const { salt, privateKey } = WalletDB.getInstance().retrieveAccount(aliasOrAddress);
+    wallet = await getSchnorrAccount(client, privateKey, deriveSigningKey(privateKey), salt).getWallet();
```

Comment:
I found this a bit hard to follow tbh, partially due to the shadowing of `salt` and `privateKey`. This seems to mean that if you pass an alias, whatever salt and pk you pass are ignored. It seems like that might be better represented with an union?

Reply:
I extracted that function into a util that will be further refine in next PRs. I also find it confusing, because that salt is referring to the one for the contract that's being deployed, not the account that's doing it (that has a hardcoded 0 in the salt). Again, this will be cleaned up, I don't think we should provide this level of refinement to end users and just offer saner defaults.